### PR TITLE
3 new and cool escape shuttles

### PR DESCRIPTION
**Pull request by airlines7**
# Document the changes in your pull request
Tired of always seeing the same 5 shuttles? Well now you can see 7 of the same shuttle!
I made 3 new escape shuttles to spice up the boredom of seeing the box shuttle (our beloved).

## Octa Prototype Shuttle
A cool doughnut shaped shuttle with (perceived) larger size
In reality its a little smaller than the box shuttle, 130~ walk-able tiles.
image

## O.C.K. Emergency Shuttle
Maints on steroids as a shuttle. OSHA approved spacecraft. 250~ walk-able tiles not taking the crates into account.
image

### The drone mafia's flagship 
180~ walk-able tiles
the drones are player controlled and have the lawset of the following. There is also a bardrone and barmaid also.
1. Be loyal to members of the organization.
2. Be rational.
3. Be a member of the team. 
4. Have class. Show hospitality to others unless they don't show class.

image

There is also a cat named Booze with 45000 hp, she's the toughest "mob" out there.
image

# Wiki Documentation

Drone mafia lore
Backstories of the escape shuttles if we do that IDK.

# Changelog

:cl:  Airlines7
rscadd: The drone mafia is now a thing
rscadd: 3 new emergency shuttles.

/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
> Octa Prototype Shuttle

Not a mapper but having lights inside of windows is a big 👎

**Comment by ynot01** [comment]
> Droni Fedora

This is going to be an admin nightmare, no thank you

Every round will consist of drones gunning down the entire crew

**Comment by airlines7** [comment]
> This is going to be an admin nightmare, no thank you

 Fine, I'll make my OWN mafia with Tommy guns and fedoras!

**Comment by ynot01** [comment]
I would much rather have an emag shuttle that just spawns random guns for the crew to have than have neutral-antags that will invariably gun down the crew every round with OP weapons

**Review by Aquizit — CHANGES_REQUESTED** [review]
Couple things:

image
This just does not look good. Would prefer walls to be continuous and not just tiles that touch diagonally.

image
This is going to kill a lot of people.

image
Assuming the Cargo Airlock button is supposed to open the blast doors in the main area, it doesn't work. 

Receiving advice about the mafia shuttle, emag shuttles are dangerous -- but it is the nature of the shuttle itself that is dangerous, not necessarily items on it. I'd say the drones are fine with a lawset that they can mess with those who break the rules, but having guns lying around isn't necessarily a good feel.

**Comment by airlines7** [comment]
image
Octa more rectangular now.

image
Don't know if you want me to add doors to these, but I'll figure something out if you want me to, walled off one door.

- Fixed the Cargo doors not working, my B on missing an underscore on the id.
- Drone mafia is snowflaked and pacified, guns are removed. They are currently invincible, but that can easily be changed if desired.
- their laws require assailants of them to be heckled, they shouldn't do any violence. They also now have the drone rule set text when spawned in.
image

**Comment by Aquizit** [comment]
Since it's an emag shuttle, the drones can do a little harm if you want them to -- maybe like stun prods for people who break the rules or some shit I dunno. But they don't need to be completely harmless *if* you want to keep the shuttle emagged. If you want to keep them pacified we can remove the emag requirement if no other dangerous stuff is involved with that shuttle.

**Comment by airlines7** [comment]
I guess I can unpacify them, just not give them guns or anything. I'll also add they should not enter CC until end of round so they don't b-line to the test area.

**Comment by Aquizit** [comment]
Yeah something like that does a bunch of stam damage or just to beat people with who break the rules. Stuff that couldn't kill too fast.

As for the CC thing shouldn't be an issue as I think if it's like the bar shuttle it doesn't pop the balloon until the shuttle is in transit.

**Comment by ynot01** [comment]
> I'll also add they should not enter CC until end of round so they don't b-line to the test area.

All IC rules are abandoned the moment the round ends, this makes no difference

**Comment by airlines7** [comment]
nah the shuttle sentience balloon pops like 2 minute before it goes to the station, they have a chance to go to the testing area and steal a pulse rifle or something. I'll figure out what weapon to give them in a few.

**Comment by airlines7** [comment]
found out pacified drones can stun time to do that

**Review by Aquizit — CHANGES_REQUESTED** [review]
image
Buttons are in the wall, should be on an adjacent tile and pixel shifted into the wall.
---
image
Missing a flash in the brig at least. Up to you if you want to also have one outside the bridge as is common in most shuttles.
---
Mafia shuttle overall needs an Emergency Shuttle Console somewhere, as every shuttle has it so a hijacker can emag/early launch. Plus if crew actually wants/needs to.

**Comment by airlines7** [comment]
> Buttons are in the wall, should be on an adjacent tile and pixel shifted into the wall.

👌

> Missing a flash in the brig at least. Up to you if you want to also have one outside the bridge as is common in most shuttles.

👌

> Mafia shuttle overall needs an Emergency Shuttle Console somewhere, as every shuttle has it so a hijacker can emag/early launch. Plus if crew actually wants/needs to.

Has one, it just looks stupid is all, I can change it to a normal computer if you'd like.
image

**Review by Aquizit — APPROVED** [review]

**Comment by Aquizit** [comment]
Maps are fine.

File: code/datums/shuttles.dm [hunk 1]
```diff
@@ -361,6 +361,32 @@
...
+	Seats and a brig space have been retrofitted to help on its current mission of saving you from the station.\
+	The higher ups complain this shuttle is very \"\ meta\"\ and \"\ increases greytide levels\"\ whatever that means."
+	admin_notes = "Has a chance to have (traitor) maint loot, you can always delete it when its at CC"
+	credit_cost = 7000
+
+/datum/map_template/shuttle/emergency/mafia
+	suffix = "mafia"
+	name = "Droni Fedora"
+	description = "I'm gonna make you an offer you can't refuse, the drone mafia has offered their 'services' to shuttle the crew.\
+	 Just be careful, if you don't show class they might heckle you. Canoli not incuded."
```
Comment (JohnFulpWillard) [inline code/datums/shuttles.dm]:
```suggestion
	description = "I'm gonna make you an offer you can't refuse, the drone mafia has offered their 'services' to shuttle the crew. \
		Just be careful, if you don't show class they might heckle you. Canoli not incuded."
```

File: code/datums/shuttles.dm [hunk 2]
```diff
@@ -361,6 +361,32 @@
...
+	While experimental, it offers great views of outside and decently stocked emergency and medical supplies."
+	admin_notes = "Doughnut yummy."
+	credit_cost = 9500 //experimental = expensive
+
+/datum/map_template/shuttle/emergency/cargo
+	suffix = "cargo"
+	name = "O.C.K. Emergency Shuttle"
+	description = "The Overnight Cargo-transport K-Class is an OSHA compliant shuttle complete with rails and warning lines to protect her crew from the cargo they are transporting.\
+	Seats and a brig space have been retrofitted to help on its current mission of saving you from the station.\
+	The higher ups complain this shuttle is very \"\ meta\"\ and \"\ increases greytide levels\"\ whatever that means."
```
Comment (JohnFulpWillard) [inline code/datums/shuttles.dm]:
```suggestion
	description = "The Overnight Cargo-transport K-Class is an OSHA compliant shuttle complete with rails and warning lines to protect her crew from the cargo they are transporting. \
		Seats and a brig space have been retrofitted to help on its current mission of saving you from the station. \
		The higher ups complain this shuttle is very \"\ meta\"\ and \"\ increases greytide levels\"\ whatever that means."
```

File: code/datums/shuttles.dm [hunk 3]
```diff
@@ -361,6 +361,32 @@
 	admin_notes = "Seriously big, even larger than the Delta shuttle."
 	credit_cost = 10000
 
+/datum/map_template/shuttle/emergency/octa
+	suffix = "octa"
+	name = "Octa Prototype Emergency Shuttle"
+	description = "Nanotrasen's experimental shuttle utilizing a unique shape to manipulate reality for a percieved larger shuttle in a smaller package. \
+	While experimental, it offers great views of outside and decently stocked emergency and medical supplies."
```
Comment (JohnFulpWillard) [inline code/datums/shuttles.dm]:
```suggestion
	description = "Nanotrasen's experimental shuttle utilizing a unique shape to manipulate reality for a percieved larger shuttle in a smaller package. \
		While experimental, it offers great views of outside and decently stocked emergency and medical supplies."
```

**Review by JohnFulpWillard — APPROVED** [review]

**Pull request merged**